org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodAnalyzer.java

```diff
@@ -86,6 +99,38 @@ public IMethodCoverage getCoverage() {
 		return coverage;
 	}
 
+	/**
+	 * {@link MethodNode#accept(MethodVisitor)}
+	 */
+	@Override
+	public void accept(final MethodNode methodNode,
+			final MethodVisitor methodVisitor) {
+		this.ignored.clear();
+		for (final IFilter filter : FILTERS) {
+			filter.filter(methodNode, this);
+		}
+
+		for (final TryCatchBlockNode n : methodNode.tryCatchBlocks) {
+			n.accept(methodVisitor);
+		}
+		for (int i = 0; i < methodNode.instructions.size(); i++) {
```

Comment:
More suggestions:

1. A reflection based assertion that `InsnList.cache` is stays `null`
2. A JaCoCo coverage check that method `InsnList.toArray()` is not executed during our tests.

But I woudn't worry too much right now - unless performance testing shows a bottle neck here.